core/src/main/java/com/orientechnologies/orient/core/index/OIndexManagerShared.java

```diff
@@ -521,7 +511,7 @@ public void run() {
         }
 
         try {
-          recreateIndexes(storage, indexesToRebuild, newDb);
+          recreateIndexes(indexesToRebuild, newDb);
         } finally {
           if (storage instanceof OAbstractPaginatedStorage) {

```

Comment:
Switch to unsafe mode and force sync is not needed anymore, let's check if this method is not used anywhere I propose to delete it.